lab-bundle-spec.md

```diff
@@ -308,20 +308,140 @@ environment:
+                Hmm. The bucket is there, but it is misconfigured.
+              es:
+                Hmm. El cubo está allí, pero está mal configurado.
+      services:
+        - target_project.StorageV1
+      code: |-
+        def check(handles, max_score)
+          storage_handle = handles['target_project.StorageV1']
+
+          # Check for bucket
+          found_bucket = ...
+          unless found_bucket
+            return { score: 0, student_message: 'bucket_missing' }
+          end
+
+          # Check bucket configuration
+          bucket_configured_correctly = ...
+          unless bucket_configured_correctly
+            return { score: 2, student_message: 'bucket_misconfigured' }
+          end
+
+          { score: max_score, student_message: 'success' }
+        end
+    - title:
+        locales:
+          en: Copy a file to the bucket
+          es: Copiar un archivo al cubo
+      maximum_score: 5
+      student_messages:
+        - success:
+            locales:
+              en:
+                Great job! You copied the file!
+              es:
+                ¡Gran trabajo! ¡Copiaste el archivo!
+        - file_missing:
+            locales:
+              en:
+                Oops! No file found.
+              es:
+                ¡Uy! No se ha encontrado el archivo.
+        - file_mismatch:
+            locales:
+              en:
+                Hmm. There's a file here, but it doesn't match the source contents.
+              es:
+                Hmm. Hay un archivo aquí, pero no coincide con el contenido de origen.
+      services:
+        - source_project.StorageV1
+        - target_project.StorageV1
+      code: |-
+        def check(handles, max_score)
+          target_storage_handle = handles['target_project.StorageV1']
+          source_storage_handle = handles['source_project.StorageV1']
+
+          # Check for file
+          found_file = ...
+          unless found_file
+            return { score: 0, student_message: 'file_missing' }
+          end
+
+          # Check file contents
+          source_contents = ...
+          target_contents = ...
+          unless source_contents == target_contents
+            return { score: 2, student_message: 'file_mismatch' }
+          end
+
+          { score: max_score, student_message: 'success' }
+        end
+```
+
+##### Code
+
+The code block must be valid Ruby code. It must have a method called `check`, and may optionally contain helper methods as well.
+
+The method `check` will be called with two arguments:
+- A map from `[RESOURCE].[SERVICE]` to a handle to that resource's service. It will only contain handles to services specified in the step's `services` array.
+- The maximum score for the step.
+
+The method `check` should return a single hash with:
+- `:score`: the number of points the student earned.
+- `:student_message`: a key from the step's `student_messages` array, which will be presented to the student in the appropriate locale.
```

Comment:
- [ ] TODO(joshgan): Student message should be passed through if it's not found in the locale hash.
- [ ] TODO(joshgan): Do we need to both `message` and `student_message`. It's unclear how `message` is currently used.